### PR TITLE
Remove potentially problematic language

### PR DESCRIPTION
**Pull request by hughsie**
Red Hat wants to drive an initiative in correcting problematic and potentially
divisive language in open source projects. These naming conventions and
descriptive phrases are hurtful and offensive to many of our colleagues across
the open source universe.

### CODE REVIEW

File: plugins/csr/fu-csr-device.c [hunk 1]
```diff
@@ -397,7 +397,7 @@ fu_csr_device_probe (FuUsbDevice *device, GError **error)
 {
 	FuCsrDevice *self = FU_CSR_DEVICE (device);
 
-	/* devices have to be whitelisted */
+	/* devices have to be quirked */
```
Comment (superm1) [inline plugins/csr/fu-csr-device.c:400]: I think instead of quirked - `allowed`

**Comment by superm1** [comment]
Are you also thinking to rename the `master` branch?

File: plugins/csr/fu-csr-device.c (hunk 1 above)
Comment (hughsie) [reply plugins/csr/fu-csr-device.c:400]: the comment is kinda obscure anyway, I've rewritten it to make sense.

**Comment by hughsie** [comment]
> Are you also thinking to rename the master branch?

Good question; I was tempted to wait until github/gitlab change the default from master to main and then change at the same time to avoid confusion. It'll also let other people fix all the tiny issues for the tools that are hardcoded to expect `master`....

**Comment by hughsie** [comment]
The one I'm not sure about is i2c master/slave. There's not really any prior art on i2c specifically, and all the suggestions make sense in a SPI-type bus but not for low-level i2c. Ideas very welcome....

**Comment by superm1** [comment]
> The one I'm not sure about is i2c master/slave. There's not really any prior art on i2c specifically, and all the suggestions make sense in a SPI-type bus but not for low-level i2c. Ideas very welcome....

I've had the same discussions with others.  There are a few things that have been suggested you can see here:
https://en.wikipedia.org/wiki/Master/slave_(technology)

I personally like primary and secondary.

**Comment by superm1** [comment]
> > Are you also thinking to rename the master branch?
> 
> Good question; I was tempted to wait until github/gitlab change the default from master to main and then change at the same time to avoid confusion. It'll also let other people fix all the tiny issues for the tools that are hardcoded to expect `master`....

I was suspecting that Github wide change will have to be opt-in in some sense because of the fact that you raised.  I see no problem changing it now though.  Realistically those types of problems don't just go away later, it just puts off fixing them to later.

**Comment by hughsie** [comment]
> I personally like primary and secondary.

That's good, but doesn't really show the `do this` relationship of master/slave. Primary/secondary sound like the devices are doing the same thing, but the primary one comes first. Same with main/secondary. tbh, it's not really the _master_ that I have problems with justifying, so maybe master/subordinate or master/serf or master/client or master/worker or master/peon or master/servant works instead.

**Comment by YehezkelShB** [comment]
If the branch name will be change from master, probably is better to remove the usage for I2C too, isn't it?
What about main/client or controller/client?

**Comment by hughsie** [comment]
> What about main/client or controller/client?

For i2c the winner seems to be initiator/responder so far.

**Comment by hughsie** [comment]
Or initiator/target -- naming is hard.

**Comment by hughsie** [comment]
See https://github.com/fwupd/fwupd/pull/2244

**Pull request merged**